Kernel/FileSystem/DevFS.cpp

```diff
@@ -270,12 +283,9 @@ KResultOr<NonnullRefPtr<Inode>> DevFSRootDirectoryInode::create_child(StringView
 
         if (auto result = new_device_inode->chmod(mode); result.is_error())
             return result;
-        if (!m_devices.try_ensure_capacity(m_devices.size() + 1))
-            return ENOMEM;
         if (!fs().m_nodes.try_ensure_capacity(fs().m_nodes.size() + 1))
             return ENOMEM;
         m_devices.append(*new_device_inode);
-        fs().m_nodes.append(*new_device_inode);
```

Comment:
Probably need to ensure I don't create a link and a device node with the same name here.